hazelcast/src/main/java/com/hazelcast/map/AbstractEntryProcessor.java

```diff
@@ -65,10 +67,21 @@ public AbstractEntryProcessor(boolean applyOnBackup) {
         return entryBackupProcessor;
     }
 
-    private class EntryBackupProcessorImpl implements EntryBackupProcessor<K, V> {
+    private class EntryBackupProcessorImpl implements EntryBackupProcessor<K, V>, HazelcastInstanceAware {
```

Comment:
That's a nice hack, but to me it begs for an explicit comment that it's the same.